src/asynqp/__init__.py

```diff
@@ -19,8 +19,12 @@
 def connect(host='localhost',
             port=5672,
             username='guest', password='guest',
-            virtual_host='/', *,
-            loop=None, sock=None, **kwargs):
+            virtual_host='/',
+            *,
+            loop=None,
+            sock=None,
+            nodelay=True,
```

Comment:
I see what you mean.

OK. So, `transport` and `protocol` are already public attributes on the `Connection`, so the client can come back and override any defaults by pulling the socket out from the transport:

```python
cn = yield from connect()
sock = cn.protocol.get('socket')
sock.setsockopt(...)
```

So you _can_ access the socket directly already to override any defaults. So perhaps the way to go is to simply set the default in `connect` (**without** a parameter to override it) and let clients change the socket after the fact. Are there any restrictions on what you can do to a socket after it's been opened?

If the client has explicitly passed in a socket (`sock is not None`) then we shouldn't change any of its options.